apps/web/src/server/app/auth/auth.guards.js

```diff
@@ -0,0 +1,75 @@
+import { checkAccessRule } from '../../lib/sso.js';
+
+/**
+ * Path to redirect to after successful login.
+ *
+ * @type {import('@pins/express').RequestHandler}
+ */
+// eslint-disable-next-line consistent-return
+export function isAuthenticated(request, response, next) {
+	if (request.session) {
+		if (!request.session.isAuthenticated) {
+			console.error('Not permitted');
+			return response.redirect('/auth/unauthorized');
+		}
+
+		next();
+	} else {
+		console.error('No session found for this request');
+		response.redirect('/auth/unauthorized');
+	}
+}
+
+export function hasAccess(options) {
+	// eslint-disable-next-line consistent-return
+	return (request, response, next) => {
+		if (request.session) {
```

Comment:
can you add a todo about introducing a session service to encapsulate auth session logic? Thanks!